applications/FluidDynamicsApplication/tests/cpp_tests/test_two_fluid_fractional_navier_stokes_element.cpp

```diff
@@ -0,0 +1,896 @@
+
+    Element::Pointer pElement = modelPart.pGetElement(1);
+
+    // Define the nodal values
+    Matrix vel_original(3, 2);
+    vel_original(0, 0) = 0.0;
+    vel_original(0, 1) = 0.1;
+    vel_original(1, 0) = 0.1;
+    vel_original(1, 1) = 0.2;
+    vel_original(2, 0) = 0.2;
+    vel_original(2, 1) = 0.3;
+
+    // Set the nodal DENSITY and DYNAMIC_VISCOSITY values
+    for (NodeIteratorType it_node = modelPart.NodesBegin(); it_node < modelPart.NodesEnd(); ++it_node)
+    {
+        it_node->FastGetSolutionStepValue(DENSITY) = pElemProp->GetValue(DENSITY);
+        it_node->FastGetSolutionStepValue(DYNAMIC_VISCOSITY) = pElemProp->GetValue(DYNAMIC_VISCOSITY);
+        it_node->FastGetSolutionStepValue(BODY_FORCE_Z) = -9.81;
+    }
+
+    for (unsigned int i = 0; i < 3; i++)
+    {
+        pElement->GetGeometry()[i].FastGetSolutionStepValue(PRESSURE) = 0.0;
+        for (unsigned int k = 0; k < 2; k++)
+        {
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY)[k] = vel_original(i, k);
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY, 1)[k] = 0.9 * vel_original(i, k);
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY, 2)[k] = 0.75 * vel_original(i, k);
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(FRACTIONAL_VELOCITY)[k] = 0.1 * vel_original(i, k);
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(FRACTIONAL_VELOCITY, 1)[k] = 0.2 * vel_original(i, k);
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(FRACTIONAL_VELOCITY, 2)[k] = 0.3 * vel_original(i, k);
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(MESH_VELOCITY)[k] = 0.0;
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(MESH_VELOCITY, 1)[k] = 0.0;
+            pElement->GetGeometry()[i].FastGetSolutionStepValue(MESH_VELOCITY, 2)[k] = 0.0;
+        }
+    }
+    pElement->GetGeometry()[0].FastGetSolutionStepValue(DISTANCE) = -1.0;
+    pElement->GetGeometry()[1].FastGetSolutionStepValue(DISTANCE) = -1.0;
+    pElement->GetGeometry()[2].FastGetSolutionStepValue(DISTANCE) = 0.5;
+
+    // Compute RHS and LHS
+    Vector RHS = ZeroVector(9);
+    Matrix LHS = ZeroMatrix(9, 9);
+
+    const auto &r_process_info = modelPart.GetProcessInfo();
+    pElement->Initialize(r_process_info); // Initialize the element to initialize the constitutive law
+    pElement->CalculateLocalSystem(LHS, RHS, r_process_info);
+
+    // Check the RHS values (the RHS is computed as the LHS x previous_solution,
+    // hence, it is assumed that if the RHS is correct, the LHS is correct as well)
+    Vector reference_RHS = ZeroVector(9);
+    reference_RHS[0] = 6075.49;
+    reference_RHS[1] = 4684.29;
+    reference_RHS[2] = -27.713;
+    reference_RHS[3] = -6671.02;
+    reference_RHS[4] = 1521.5;
+    reference_RHS[5] = -16.8578;
+    reference_RHS[6] = -241.804;
+    reference_RHS[7] = -4409.55;
+    reference_RHS[8] = -5.5792;

```

Comment:
See how I do this in my tests. It simplifies the eventual need of changing the values.

Reply:
Perfect! Modified